Sources/TuistGenerator/Generator/DerivedFileGenerator.swift

```diff
@@ -0,0 +1,124 @@
+import Basic
+import Foundation
+import TuistCore
+import XcodeProj
+
+protocol DerivedFileGenerating {
```

Comment:
👍 this is great!

We can re-use this when continuing work on static frameworks resources to generate the `Bundle.swift` :)

Reply:
Yeah! That's the idea. I think generating resources and code on the fly will show up in other features so I thought about having a component that helps with that.